src/extension/app/utils/admin-client.js

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { callAdmin, createAdminUrl } from '../../utils/admin.js';
+
+/**
+ * @typedef {import('../store/app.js').AppStore} AppStore
+ */
+
+/**
+ * @typedef {import('../store/site.js').SiteStore} SiteStore
+ */
+
+/**
+ * Facilitates calls to the [AEM Admin API]{@link https://www.aem.live/docs/admin.html}
+ * and displays localized error messages.
+ */
+export class AdminClient {
+  /**
+   * Enumeration of rate limiting sources.
+   * @private
+   * @type {Object<string, string>}}
+   */
+  RATE_LIMITER = {
+    NONE: '',
+    ADMIN: 'AEM',
+    ONEDRIVE: 'Microsoft SharePoint',
+  };
+
+  /**
+   * The app store.
+   * @private
+   * @type {AppStore}
+   */
+  appStore;
+
+  /**
+   * The site store
+   * @private
+   * @type {SiteStore}
+   */
+  siteStore;
+
+  /**
+   * Returns an Admin Client.
+   * @param {AppStore} appStore The app store
+   */
+  constructor(appStore) {
+    this.appStore = appStore;
+    this.siteStore = appStore.siteStore;
+  }
+
+  /**
+   * Returns the action name based on API and nature of request.
+   * @private
```

Comment:
I started out with `#` and to be honest, I find it makes the code a bit harder to read. I also got the impression that you weren't a huge fan. My understanding is that `@private` only prevents the member from showing up in the IDE's auto-complete (which was my main concern here) while `#` also makes it impossible to use from the outside the class (actual security). I'm open to switching to `#` syntax if you prefer it.

Reply:
Here's what it would look like: https://github.com/adobe/aem-sidekick/compare/centralized-admin...centralized-admin-private-members?expand=1#diff-85dbd67e48b186c9fe814c214ea5893590052de3a840e91dc4cf9e152ab72d85

Reply:
I think I'm warming up to it and it's probably the right thing to be doing.. I agree that we mostly only care about it not showing up in the IDE vs building a library and worrying about people breaking things. We are also building for `chrome 64` which came out in 2017 so it won't make it into the bundle in the end.

I was more curious where you head is at on it.. I think it's a commitment and if we decide to use it we should adopt it everywhere.

Reply:
I still feel the syntax makes the code less readable, but at the same time I also agree it's probably the right thing to do. I'd propose to tackle this in a separate, more broadly scoped PR. I created #151 to track it.